src/twisted/web/_flatten.py

```diff
@@ -178,7 +177,7 @@ def escapedComment(data: Union[bytes, str]) -> bytes:
     """
     if isinstance(data, str):
         data = data.encode("utf-8")
-    data = data.replace(b"--", b"- - ").replace(b">", b"&gt;")
+    data = data.replace(b"--", b"- - ")
```

Comment:
This is still mangling, not quoting (it looks like quoting is not possible here) but it is a less legible form of quoting.

What do you think about this, instead?

```suggestion
    data = data.replace(b"-->", b"--&gt;")
```

I parsed a `--` in a comment with `html5lib` 1.1 and despite the spec's prohibition on `--`, this document:

```html
<!DOCTYPE html>
<html>
 <head>
  <title>hi</title>
 </head>
 <body>
  <!-- -- -->
 </body>
</html>
```

responds to this program:
```python3
from html5lib import parse
from xml.etree.ElementTree import tostring

with open("comment.html") as f:
    d = parse(f, namespaceHTMLElements=False)
    c = list(d.iter())[-1]
    print(c.text)
```
by printing out `--`, and can happily re-serialize it without any warnings I can detect.

Given that we need to mangle it, it would seem to me that we should mangle as minimally as possible, which appears to be what browsers and reference specs do. For example, that HTML file in a browser looks like this as well:

```
> document.body.childNodes[1].textContent
< " -- "
```

Most of all, adding in a space to mangle like this seems really arbitrary.

Reply:
Awesome @glyph, thank you for your explanation, it was incredibly helpful and improved my understanding. I also appreciate the invocation of Greek mythology to help break the impasse. Alexander would be proud.   

Also, while it's minor issue, I followed your lead and ran similar tests and noticed that ending a comment with a dash does not appear to have any effect. Following the ideal to perturb as little as possible, I would be tempted to remove the following code block, but I don’t want to be petty and make any unwarranted changes, so I won't touch it unless directed to do so.
``` 
if data and data[-1:] == b"-": 
    data += b" " 
``` 

I'll jump on this as soon as I get a chance to do so. Once again I appreciate your input on this.

Reply:
Oh, that's a great catch. Thank you for pointing it out :-).

Reply:
(And yes, I agree that adding whitespace to break the connection of the ending `-` with the closing `-->` is probably the least intrusive thing I can think of.)